File: generated-code/scala/codecraft/TcpReadWrite/src/main/scala/trans_gen_test/model/RepairAction.scala
package trans_gen_test.model

import trans_gen_test.util.StreamUtil

/**
 * Repair action
 *
 * @param target Target entity's ID
 */
case class RepairAction(target: Int) {
    /**
     * Write RepairAction to output stream
     */
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, target)
    }

    /**
     * Get string representation of RepairAction
     */
    override def toString(): String = {
        var stringBuilder = new StringBuilder("RepairAction { ")
        stringBuilder.append("target: ")
        stringBuilder.append(target)
        stringBuilder.append(" }")
        stringBuilder.toString()
    }
}

object RepairAction {
    /**
     * Read RepairAction from input stream
     */
    def readFrom(stream: java.io.InputStream): RepairAction = RepairAction(
        StreamUtil.readInt(stream)
    )
}

File: generated-code/java/codecraft/FileReadWrite/src/main/java/trans_gen_test/model/RepairAction.java
package trans_gen_test.model;

import trans_gen_test.util.StreamUtil;

/**
 * Repair action
 */
public class RepairAction {
    /**
     * Target entity's ID
     */
    private int target;

    /**
     * Target entity's ID
     */
    public int getTarget() {
        return target;
    }

    /**
     * Target entity's ID
     */
    public void setTarget(int value) {
        this.target = value;
    }

    public RepairAction(int target) {
        this.target = target;
    }

    /**
     * Read RepairAction from input stream
     */
    public static RepairAction readFrom(java.io.InputStream stream) throws java.io.IOException {
        int target;
        target = StreamUtil.readInt(stream);
        return new RepairAction(target);
    }

    /**
     * Write RepairAction to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, target);
    }

    /**
     * Get string representation of RepairAction
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("RepairAction { ");
        stringBuilder.append("target: ");
        stringBuilder.append(String.valueOf(target));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}